src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Call.cs

```diff
@@ -364,11 +376,219 @@ private static bool IsSafeForReordering(BoundExpression expression, RefKind kind
+            {
+                Debug.Assert(temps.Count > 0);
+                Debug.Assert(sideEffectsBuilder!.Count > 0);
+                Debug.Assert(!positionsAssignedToTemp.IsNull);
+                Debug.Assert(receiverStored == requiresInstanceReceiver);
+
+                // If this was an interpolated string conversion that needed a receiver, the
+                // first store is actually the receiver, not the argument, so turn the receiver
+                // into a sequence. Otherwise, turn the first actual argument into that sequence.
+
+                var sideEffectsResult = requiresInstanceReceiver ? rewrittenReceiver : visitedArgumentsBuilder[0];
+                Debug.Assert(sideEffectsResult != null);
+
+                var sideEffectsSequence = _factory.Sequence(
+                    locals: ImmutableArray<LocalSymbol>.Empty,
+                    sideEffects: StaticCast<BoundExpression>.From(sideEffectsBuilder.ToImmutableAndFree()),
+                    sideEffectsResult);
+
+                if (requiresInstanceReceiver)
+                {
+                    rewrittenReceiver = sideEffectsSequence;
+                    receiverIsArgumentSideEffectSequence = true;
+                }
+                else
+                {
+                    visitedArgumentsBuilder[0] = sideEffectsSequence;
+                    positionsAssignedToTemp[0] = false;
+                }
+
+                if (refKindsBuilder != null)
+                {
+                    argumentRefKindsOpt = refKindsBuilder.ToImmutableAndFree();
+                }
+            }
+
+            return visitedArgumentsBuilder.ToImmutableAndFree();
+
+            InterpolatedStringHandlerData? setupInterpolationDataIfNeeded(
+                int argumentIndex,
+                ImmutableArray<RefKind> argumentRefKindsOpt,
+                ref bool receiverStored,
+                ref int lastStoredIndex,
+                ref BoundExpression? visitedReceiver,
+                ref ArrayBuilder<LocalSymbol>? locals,
+                ref ArrayBuilder<BoundExpression>? sideEffectsBuilder,
+                ref BitVector positionsAssignedTemps)
+            {
+                var argument = arguments[argumentIndex];
+
+                if (argument is BoundConversion { ConversionKind: ConversionKind.InterpolatedStringHandler, Operand: BoundInterpolatedString operand })
+                {
+                    // Handler conversions are not supported in expression lambdas.
+                    Debug.Assert(!_inExpressionLambda);
+                    var interpolationData = operand.InterpolationData.GetValueOrDefault();
+                    var creation = (BoundObjectCreationExpression)interpolationData.Construction;
+
+                    if (creation.Arguments.Length > (interpolationData.HasTrailingHandlerValidityParameter ? 3 : 2))
+                    {
+                        Debug.Assert(!((BoundConversion)argument).ExplicitCastInCode);
```

Comment:
>Debug.Assert(!((BoundConversion)argument).ExplicitCastInCode); [](http://example.com/codeflow?start=24&length=62)

Is explicit cast fine when condition of the enclosing ```if``` isn't met?

Reply:
Yes, because you could explicitly do `(InterpolatedStringHandlerType)$""`, which will be an explicit cast in code with either 2 or 3 arguments.